engine/src/main/java/org/camunda/bpm/engine/TelemetryService.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.camunda.bpm.engine;
+
+import org.camunda.bpm.engine.impl.telemetry.dto.TelemetryData;
+
+/**
+ * Service which provides access to data collected for telemetry.
+ * Telemetry data is also available if you did not opt-in to share the data with Camunda.
+ */
+public interface TelemetryService {
+
+  /**
+   * Retrieves and returns the current telemetry data.
+   */
+  TelemetryData getData();
```

Comment:
This API should not return impl classes and it should define interfaces with proper Javadocs for what is returned.